src/engine/qcommon/files.cpp

```diff
@@ -637,9 +637,22 @@ void FS_LoadBasePak()
 
 void FS_LoadAllMapMetadata()
 {
+	const std::string pak_map_prefix("map-");
+
 	for (const auto& pak: FS::GetAvailableMapPaks()) {
```

Comment:
GetAvailableMapPaks is already supposed to filter for paks that start with `map-`, but the filtering is disabled when fs_legacypaks is on. So either this function should be changed to always return only paks starting with `map-`, or the behavior here should do something different when the map does not start with `map-`.

Reply:
This is to load meta information to build the create server ui.

We may have to keep the test for the `map-` prefix in both.

This one is to filter maintained maps that were packaged the right we, we are not expecting legacy paks to have meta information. So, since `GetAvailableMapPaks` may list all map packs even the legacy ones, we have to filter in `FS_LoadAllMapMetadata`

We can either drop the `map-` prefix test in `GetAvailableMapPaks`, but we can keep it for the sole purpose of speeding-up things for when support for legacy packs is disabled.

The `map-` prefix for maps is part of DPK format definition, so this is OK to keep the test for it in engine wen looking for map packs when support for legacy packs is disabled.

That `FS_LoadAllMapMetadata` function really looks like a game-specific feature, especially Unvanquished specific feature. We may imagine other games doing things differently than the `meta/` thing (even if we strongly recommend to do so, as best practices and best support). Or we have to decide the meta thing is part of the DPK format.

At this point it's fairly legit to think a game may extends `FS_LoadAllMapMetadata`, extending the existing meta stuff with checks for other games (like legacy `.arena` files). It's not up to the engine to support all existing features.

Also, we can imagine a game wanting to support more formats would rely on this unmodified `FS_LoadAllMapMetadata` but add another function to complete the list with other formats. In this case we may remove the `map-` prefix and make sure this function is never used for other formats.